src/GNDStk/gnds-tree-visit.hpp

```diff
@@ -0,0 +1,126 @@
+
+// -----------------------------------------------------------------------------
+// Definitions:
+//    Tree::meta()
+//    Tree::child()
+//    Tree::operator()()
+// -----------------------------------------------------------------------------
+
+// ------------------------
+// Tree::meta()
+// ------------------------
+
+// string
+template<
+   template<class...> class MCON,
+   template<class...> class CCON
+>
+inline decltype(auto) Tree<MCON,CCON>::meta(const std::string &str) const
+{
+   assert(!empty());
+   return root->meta(str);
+}
+
+// meta_t
+template<
+   template<class...> class MCON,
+   template<class...> class CCON
+>
+template<class T>
+inline decltype(auto) Tree<MCON,CCON>::meta(const meta_t<T> &kwd) const
+{
+   assert(!empty());
+   return root->meta(kwd);
+}
+
+
+// ------------------------
+// Tree::child()
+// ------------------------
+
+// string
+template<
+   template<class...> class MCON,
+   template<class...> class CCON
+>
+inline decltype(auto) Tree<MCON,CCON>::child(const std::string &str) const
+{
+   assert(!empty());
+   return root->child(str);
+}
+
+// child_t
+template<
+   template<class...> class MCON,
+   template<class...> class CCON
+>
+template<class T, class META, class CHILD>
+inline decltype(auto) Tree<MCON,CCON>::child(
+   const child_t<T,META,CHILD> &kwd
+) const {
+   assert(!empty());
+   return root->child(kwd);
```

Comment:
What is `root` here? I presume that it is a member of the `Tree<...,...>` class, but I'm not sure because I don't know where the `Tree` class is defined.

Reply:
Top pointer to initial tree node. "root" is somewhat of a standard name for this sort of thing in a tree structure. I could call it "top" or some such thing. I may also see if I can consolidate some of the use of "root" into other things, and possibly be able to make it a private data member. I've already done a bit of this, by using "not tree.empty()" in place of "root != nullptr" in the places you mentioned.